calico/felix/fetcd.py

```diff
@@ -917,18 +979,44 @@ def parse_endpoint(config, combined_id, raw_json):
     return endpoint
 
 
-def parse_rules(profile_id, raw_json):
+def parse_tier_data(tier, data):
+    data = safe_decode_json(data, log_tag="tier %s" % tier)
+    try:
+        common.validate_tier_data(tier, data)
+    except ValidationFailed as e:
+        _log.error("Validation failed for tier data for tier %s: %r",
+                   tier, e)
+        return None
+    else:
+        return data
+
+
+def parse_profile(profile_id, raw_json, require_selector=False,
+                  require_order=False):
     rules = safe_decode_json(raw_json, log_tag="rules %s" % profile_id)
     try:
-        common.validate_rules(profile_id, rules)
+        common.validate_profile(profile_id, rules)
     except ValidationFailed as e:
-        _log.exception("Validation failed for profile %s rules: %s; %r",
-                       profile_id, rules, e)
+        _log.exception("Validation failed for profile %s rules: %s",
+                       profile_id, rules)
         return None
     else:
         return rules
 
 
+def parse_policy(profile_id, raw_json, require_selector=False,
```

Comment:
Similarly here - except I would expect that require_selector is always True for policies.